navigation/MainNavigator.tsx

```diff
@@ -35,19 +51,55 @@ const MainNavigator: FunctionComponent = () => {
     if (navigationState.type == NavigationScreens.intro) {
       return <MainScreen />;
     } else if (navigationState.type === "recordEpisodes") {
-      // Putting a default value into the overlay creator until we have the actual value
       return (
         <CameraFlowNavigator
           objects={navigationState.episodes}
-          overlayCreator={(episode) => (
-            <Text style={{ color: "white" }}>{episode.name}</Text>
+          overlayCreator={(episode: Episode, index: number) => (
+            <EpisodeOverlay episode={episode} index={index} />
           )}
+          nameCreator={(episode: Episode, index: number) =>
+            `${participantId}/${participantId}_${recordingDay}_${index}`
+          }
+          nextState={{ type: NavigationScreens.predictions }}
+          recordingDay={recordingDay}
         />
       );
     } else if (navigationState.type == NavigationScreens.createEpisode) {
       return <EpisodeNavigator />;
     } else if (navigationState.type == NavigationScreens.predictions) {
       return <EpisodePredictionWrapper />;
+    } else if (
```

Comment:
This conditional is getting pretty long and it's only going to get longer, is there no better way to do this? I don't have any solutions, but just wondering.

Reply:
Hmmm good question. Super open to suggestions here, but for right now will leave as is.